client/lib/feed-stream-store/paged-stream.js

```diff
@@ -249,16 +249,19 @@ export default class PagedStream {
 	}
 
 	dismissPostsFromSite( blogId ) {
-		const removed = remove( this.postKeys, { blogId } );
-		if ( removed.length === 0 ) {
-			return; // nothing changed
+		let indexToRemove;
```

Comment:
though we have to change the postKeys array to a new instance. Otherwise the shallow equals / pure components we use later on will fail to see the update.